server/tracedb/client/http.go

```diff
@@ -0,0 +1,123 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
```

Comment:
Generic http client

Reply:
Is it worth adding a generic http client? Every api is different and I can't see a benefit in having a generic client for it. I think we would have to make it extremely configurable, and I don't think that's worth the effort.

This might be applicable to the gRPC client too.

@schoren @danielbdias I want your thoughts here

Reply:
The thing about having it extremely configurable is that the complexity of the generic client will explode in a little time (for instance, having a lot of `ifs` and `options`), that's correct?

To tame that, I was thinking if we can follow something like the [strategy design pattern](https://sourcemaking.com/design_patterns/strategy): we could have one client for each DataStore and the data store itself decides how to use communication strategy internally (HTTP and/or gRPC).

Reply:
@danielbdias the initial idea was to build something similar to the strategy pattern, but as this is a double layer (trace db + clients) I think a better way to describe it is as an [Abstract Factory](https://refactoring.guru/design-patterns/abstract-factory).

The situation here is that tracedbs are all similar until the `getTraceById` step, where each one has a different way of handling that. I can create a `data_store` interface so each of the clients has the same implementation, and a single callback type

Reply:
I updated the code based on the comments, let me know what you think